shared/types/form/form.ts

```diff
@@ -150,6 +150,8 @@ export interface FormBase {
   webhook: FormWebhook
 
   responseMode: FormResponseMode
+
+  newEncryptionBoundary: boolean
```

Comment:
For this, as the encryption boundary shift only applies to storage mode, should we move this under StorageFormBase instead?

Reply:
On further thought, i think it is ok since it is an internal flag. We may want to test sending email mode forms to the new non-encrypted endpoint, for debugging purpose.

Reply:
Hmm good point for shifting it to `StorageFormBase`. What do you mean by debugging purposes using email mode forms actually? Do you mean for virus scanning? 
Side question: for virus scanning of attachments in email mode forms, are we currently dependent on the virus scanning by email clients?

Reply:
I was thinking that in the process of debugging, we might want to send an email mode form to the non encrypted mode form endpoint. so wasn't too fixed on necessarily moving it to `StorageFormBase`

yes, there is virus scanning by email clients atm, so virus scanning for email mode is not a priority. we could implement it later for defence in depth

Reply:
Okay I see what you mean 🤔 Not sure if using email mode forms to debug the new storage mode endpoint will be very useful since there are things like the public key which is required by storage mode forms but not email mode forms though.

Reply:
Got it; fine to shift it to `StorageFormBase` in that case. Don't feel strongly eitherway

Reply:
Instead of shifting it completely, I set it as optionally `never` type for email mode instead, similar to the `publicKey` prop in the schemas. Doing this instead because the types `IFormDocument` and `IPopulatedForm` used in `getFilteredResponses` and `checkNewBoundaryEnabled` respectively need to access the flag.
Done in [175f4ea](https://github.com/opengovsg/FormSG/pull/6595/commits/175f4ea3f85645beb18f823daf58c05878d63185) and resulting broken tests fixed in [e943387](https://github.com/opengovsg/FormSG/pull/6595/commits/e94338770bca4d7b457a4adab0b1340d03e8846e).